src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs

```diff
@@ -71,5 +72,32 @@ public static void AssertRoundtrips<T>(T expected, params Func<T, object>[] addi
                 }
             }
         }
+
+        public static void AssertExceptionDeserializationFails<T>() where T : Exception
+        {
+            // .NET Core throws a PlatformNotSupportedException for deserializing many exceptions.
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
+            {
+                return;
+            }
+
+            var info = new SerializationInfo(typeof(T), new FormatterConverter());
```

Comment:
Because the constructors call the base class constructor, which calls the base class constructor and so on. This is before the body of the constructor has been executed. Therefore, we end up calling the serialization ctor for Exception before get to the block which throws. Exception's ctor validates the bucket so we need a complete bucket
7